src/js/node/fs.promises.ts

```diff
@@ -89,11 +89,37 @@ function watch(
   };
 }
 
+let lazy_cp: typeof import("../internal/fs/cp").default | null = null;
+async function cp(src: string | URL, dest: string | URL, options): Promise<void> {
+  // attempt to use the native code version if possible
+  // and on MacOS, simple cases of recursive directory trees can be done in a single `clonefile()`
+  if (!options) return fs.copyFile(src, dest);
+  if (typeof options !== "object") {
+    throw new TypeError("options must be an object");
+  }
+  let mode = options.mode;
+  if (
+    options.dereference ||
+    options.filter ||
+    options.preserveTimestamps ||
+    options.verbatimSymlinks ||
+    !options.force ||
```

Comment:
`force` or `errorOnExist` will usually not be set, which means the fast path will be the unusual path.

Maybe `options` should be combined to have the default values set to distinguish between when options was set to an empty object/missing some properties versus when the developer explicitly passed `force: false` or `errorOnExist: false`.